src/mistral_common/protocol/instruct/normalize.py

```diff
@@ -90,18 +91,48 @@ def _normalize_json_content(self, content: Optional[str]) -> str:
             normalized_content = content
         return normalized_content
 
-    def _aggregate_content_chunks(self, content: Union[str, List[TextChunk]], chunk_join_str: str = "\n\n") -> str:
+    @overload
+    def _aggregate_content_chunks(
+        self, content: List[Union[str, TextChunk, ThinkChunk]], chunk_join_str: str = "\n\n"
+    ) -> Union[str, List[Union[TextChunk, ThinkChunk]]]: ...
+    @overload
+    def _aggregate_content_chunks(self, content: str, chunk_join_str: str = "\n\n") -> str: ...
+    @overload
+    def _aggregate_content_chunks(self, content: List[str], chunk_join_str: str = "\n\n") -> str: ...
+    def _aggregate_content_chunks(
+        self, content: Union[str, List[Union[str, TextChunk, ThinkChunk]], List[str]], chunk_join_str: str = "\n\n"
+    ) -> Union[str, List[Union[TextChunk, ThinkChunk]]]:
         if isinstance(content, list):
-            return chunk_join_str.join([chunk.text for chunk in content])
-        else:
+            aggregated_content: List[Union[TextChunk, ThinkChunk]] = []
+            for chunk in content:
+                if isinstance(chunk, str):
+                    chunk = TextChunk(text=chunk)
+
+                if isinstance(chunk, TextChunk):
+                    if aggregated_content and isinstance(aggregated_content[-1], TextChunk):
+                        aggregated_content[-1].text += chunk_join_str + chunk.text
+                    else:
+                        aggregated_content.append(chunk)
+                elif isinstance(chunk, ThinkChunk):
```

Comment:
what if two think chunks follow each other - is this allowed? Should we maybe throw in this case?